Add bare metal infra configuration

minmzzhang opened the pull request.

sunya-ch commented on infrastructure-preparation/README.md:
```diff
@@ -0,0 +1,112 @@
+# Multi-NIC CNI Infrastructure Support
+
+Multi-NIC CNI supports three different infrastructure configurations, each with its own specific features and requirements.
```
Can we have table of content here?

minmzzhang replied:
sure!

sunya-ch commented on infrastructure-preparation/README.md:
````diff
@@ -0,0 +1,112 @@
+- Appropriate IAM permissions
+
+### Configuration
+```yaml
````
Why yaml format here?

minmzzhang replied:
should add files' actual link here

sunya-ch commented on infrastructure-preparation/README.md:
````diff
@@ -0,0 +1,112 @@
+- Worker node access
+
+### Configuration
+```hcl
````
Why hcl?

minmzzhang replied:
should add files' actual link here

minmzzhang commented:
Hi @sunya-ch , I updated the README.md, added the ipvlan configuration yaml file under bare-metal and rearranged the instruction, please help review, thank you!

sunya-ch commented on infrastructure-preparation/bare-metal/README.md:
```diff
@@ -0,0 +1,85 @@
+# OpenShift Infrastructure Preparation
```
```suggestion
# OpenShift on Bare Metal Infrastructure Preparation
```

Note, and the other `README.md` should follow the same pattern.

sunya-ch reviewed (approved):
Thank you, LGTM.
This PR has completed on add bare metal infra configuration.
However, still need later PR to update the README.md on aws and ibmcloud to the same format.
